alchemy-core/src/main/java/io/rtr/alchemy/models/Allocations.java

```diff
@@ -120,16 +113,16 @@ private void mergeAdjacentAllocations() {
      * @param treatment The treatment
      * @param size The number of bins
      */
-    public void allocate(Treatment treatment, int size) {
-        Preconditions.checkArgument(
-                getUnallocatedSize() >= size,
-                "not enough free bins to allocate treatment %s with size %s given %s unallocated bin(s)",
-                treatment.getName(),
-                size,
-                getUnallocatedSize());
+    public void allocate(final Treatment treatment, final int size) {
+        if (getUnallocatedSize() < size) {
```

Comment:
Is this the right logic? I see unallocatedSize() >= size in the previous version

Reply:
Yes, `Preconditions` need "how it should be" as an argument.